This patch taken from the net-snmp community fixes an issue related to mismatch of the 
indices in the host resources tables for monitoring disk storage.
 
The details for the upstream bug/patch are given below.
BUG: 1557372: Realign hrFSStorageIndex with hrStorageTable
http://net-snmp.svn.sourceforge.net/viewvc/net-snmp?view=revision&revision=17138

--- net-snmp-5.4.1.old/agent/mibgroup/host/hr_filesys.c	2014-03-25 07:06:17.172039070 -0400
+++ net-snmp-5.4.1/agent/mibgroup/host/hr_filesys.c	    2014-03-25 07:10:05.467869740 -0400
@@ -4,6 +4,9 @@
  */
 
 #include <net-snmp/net-snmp-config.h>
+#include <net-snmp/net-snmp-includes.h>
+#include <net-snmp/agent/net-snmp-agent-includes.h>
+#include <net-snmp/agent/hardware/memory.h>
 #include "host_res.h"
 #include "hr_filesys.h"
 #include "hr_storage.h"
@@ -546,7 +549,7 @@
             long_return = 2;    /* others probably aren't */
         return (u_char *) & long_return;
     case HRFSYS_STOREIDX:
-        long_return = fsys_idx + HRS_TYPE_FIXED_MAX;
+        long_return = fsys_idx + NETSNMP_MEM_TYPE_MAX;
         return (u_char *) & long_return;
     case HRFSYS_FULLDUMP:
         return when_dumped(HRFS_entry->HRFS_name, FULL_DUMP, var_len);
